src/ProspectClient.php

```diff
@@ -0,0 +1,25 @@
+<?php
+namespace Silktide\ProspectClient;
+
+use Silktide\ProspectClient\Api\ReportApi;
+use Silktide\ProspectClient\Http\HttpWrapper;
+
+class ProspectClient
+{
+    private HttpWrapper $httpWrapper;
+    private ReportApi $reportApi;
+
+    public function __construct(string $apiKey, HttpWrapper $httpWrapper = null)
+    {
+        $this->httpWrapper = $httpWrapper ?? new HttpWrapper($apiKey);
+    }
+
+    public function getReportApi():ReportApi
+    {
+        if(!isset($this->reportApi)) {
```

Comment:
Unless there are real performance considerations, it would be cleaner without the additional caching.

Services as a general rule should not have state and should have a good reason before it has it added :P

Reply:
I don't know why I did it this way. Doesn't seem to add any benefit. I even looked at your branch to see if I could blame you for the idea... but no it's just me being dumb.